frontend/src/main/scala/bloop/engine/caches/StateCache.scala

```diff
@@ -24,7 +24,7 @@ final class StateCache(cache: ConcurrentHashMap[AbsolutePath, State]) {
    * @return The passed `State`, with a status code of `Ok`.
    */
   def updateBuild(state: State): State = {
-    cache.put(state.build.origin, state)
+    Option(cache.put(state.build.origin, state)).getOrElse(state)
```

Comment:
Did you ever hit `null` here? Wouldn't that be a scary bug ???

Reply:
Also it looks like the java doc is not in sync with the code, but it already wasn't...

Reply:
Not scary in the hand, this happens in tests with a hand-rolled state